### PR TITLE
feat(retrospect): add post-compaction receipt gate (#600)

### PR DESCRIPTION
**Pull request by devseunggwan**
## Summary

The Stage 2 "Compaction + readable transcript (MUST)" prose in the retrospect
skill recurred as a salient-window default after it shipped — rule exists ≠
retrieval. Two same-day, independent retrospect sessions scoped friction to the
compaction summary's salient window and only enumerated the full transcript
after a user challenge. Prose alone does not gate control flow.

This adds **Gate-7**, a session-level Stop-hook structural check that turns the
prose MUST into a machine-checkable receipt — the durable enforcement rung the
prose lacked.

## How it works

When the transcript contains a compaction marker (`"isCompactSummary":true`),
the Stage 3 report must carry a `retrospect:transcript_receipt` fence holding
the real friction-scan command output (is_error / user-turn / interrupt counts),
not a self-authored count. Absent the fence, Stage 3 is blocked with a guidance
message. A `_skipped` variant covers the legitimate no-compaction-in-window case.
The structural marker is an HTML comment so it survives the hook's code-fence
stripping; the real command output lives inside the fenced block.

## Changes

- `skills/retrospect/SKILL.md`: receipt requirement at Stage 2, fence definition
  (Stage 3 output §3e), AUTHORITATIVE_SCHEMA co-update entry (4), carve-out
  EXCEPTION note, two failure-mode rows.
- `hooks/completion-verify/retrospect-mix-check/impl.sh`: Gate-7 detection
  (whole-file compaction grep + receipt-fence presence) + block reason.
- tests: 4 cases (block-no-receipt / pass-with-receipt / pass-no-compaction
  regression / pass-skip-variant).

## Verification

- `bash scripts/run-tests.sh` — ALL TESTS PASSED (46 cases + 11 fixtures,
  manifest check, repo-agnosticism)
- `shellcheck --severity=warning` on modified files — clean
- `ruff check .` — clean

Pre-commit verified: bash scripts/run-tests.sh — ALL TESTS PASSED (46 cases + 11 fixtures, manifest check OK, repo-agnosticism PASS); shellcheck --severity=warning on modified files clean; ruff check . clean

Refs #600 (follow-up: mechanizes the prose MUST shipped in #600 into a hook gate)

Not verified: a live post-compaction retrospect run (unit-test coverage only).

## Summary by CodeRabbit

* **New Features**
  * Introduced Gate-7 enforcement: post-compaction sessions must include a transcript receipt fence or a valid skipped-state marker; otherwise Stage 3 is blocked.

* **Documentation**
  * Updated blocking guidance with Gate-7 remediation and receipt-fence handling instructions.

* **Tests**
  * Added scenarios validating Gate-7 enforcement, bypass conditions, skipped-state handling, and compaction marker edge cases.

### CODE REVIEW

File: hooks/completion-verify/retrospect-mix-check/impl.sh [hunk 1]
```diff
@@ -233,6 +241,25 @@ while IFS= read -r row; do
...
+# compaction summary's salient narrative instead of enumerating the transcript
+# (skills/retrospect/SKILL.md Stage 2 "Compaction + readable transcript MUST").
+# The skipped variant ('retrospect:transcript_receipt_skipped') matches the same
+# substring and satisfies the structural check; SKILL.md governs when that
+# variant is legitimate. The compaction grep scans the WHOLE file (not the tail)
+# because the marker can sit far back in a long post-compaction session.
+GATE7_VIOLATION=""
+if grep -Eq '"isCompactSummary"[[:space:]]*:[[:space:]]*true' "$TRANSCRIPT_PATH" 2>/dev/null; then
+  if ! printf '%s\n' "$MOST_RECENT_BLOCK" | grep -qF 'retrospect:transcript_receipt'; then
+    GATE7_VIOLATION="post-compaction session but the Stage 3 report has no '<!-- retrospect:transcript_receipt begin/end -->' fence — run the full-transcript friction scan and paste the REAL command output (is_error / user-turn / interrupt counts) in the receipt fence before Stage 3"
```
[the remark this answers is not in the document]
Comment (devseunggwan) [reply hooks/completion-verify/retrospect-mix-check/impl.sh]: Addressed in 7657859 — but not by switching to `jq`. Here is the reasoning.

**The premise was empirically falsified.** On a transcript saturated with textual mentions of the marker, the bare pattern matched exactly the same lines as structural `jq` (grep count == jq count). A textual mention is JSON-escaped to `\"isCompactSummary\":true` on disk, and the unescaped-quote pattern never matches the escaped form — only a real top-level field, whose quote is preceded by `{` or `,` (never a backslash), matches.

**`jq` was deliberately not adopted.** The transcript is appended live, and this Stop hook can run while the final line is still being written. `jq` aborts on a partially-written / malformed line; `grep` tolerates it. Switching would trade a false-positive that does not occur for a real abort risk.

**What changed instead:** the pattern is hardened with a `(^|[^\\])` prefix so the escaped-form exclusion is explicit rather than relying on a reader knowing the escaping invariant, and a `G5` regression test locks the no-false-trigger property. (In fairness, `G5` passes under both the old and new patterns — it is a property lock, not proof of a behavior change.)

File: skills/retrospect/SKILL.md [hunk 2]
```diff
@@ -859,7 +861,7 @@ If neither is present → return the finding to **Gate-1 / step 4** for label re
...
...ipped). `gate_5_verdict` is informational-only and INTENTIONALLY EXCLUDED from Stop hook parsing — the hook silently ignores it; Gate-5 enforcement is procedural inside Stage 2.5. (Stop hook wiring for Gate-5 is a deferred follow-up similar to Gate-4's progression in PR #340 — file a follow-up issue when procedural compliance proves insufficient.) `gate_6_verdict` is likewise informational-only and INTENTIONALLY EXCLUDED from Stop hook parsing (parallel to `gate_3_verdict` and `gate_5_verdict`) — the hook silently ignores it; Gate-6 enforcement is procedural inside Stage 2.5. **Gate-7 (Transcript-Enumeration Receipt)** is a session-level structural check (NOT a per-finding Stage 2.5 gate): when the transcript contains a compaction marker (`"isCompactSummary":true`), the Stop hook requires the `retrospect:transcript_receipt` fence (Stage 3 output §3e) in the report and blocks Stage 3 when it is absent. Unlike the informational fences below, `transcript_receipt` **IS** parsed by the hook — it is the structural enforcement of the Stage 2 "Compaction + readable transcript (MUST)" prose, added because that prose recurred as a salient-window default after shipping (issue #600 follow-up).
```
[the remark this answers is not in the document]
Comment (devseunggwan) [reply skills/retrospect/SKILL.md]: Fixed in aefe88f — all five references in the contract prose now point to `hooks/completion-verify/retrospect-mix-check/impl.sh` and `tests/hooks/completion-verify/test_retrospect_mix_check.sh`. Line 258 already used the correct path, so the file was internally inconsistent; it is now uniform.

File: skills/retrospect/SKILL.md [hunk 3]
```diff
@@ -953,7 +967,7 @@ Stage 3 output MUST emit, in this order:
    - `Proposed Actions (1~2)`: comma-separated subset of `memory`, `issue`, `claude_md_draft`, `skill_idea`, `hook_code`, `upstream_feedback`; for findings marked `external=true` by Stage 2.5 Gate-4, append ` (external)` suffix to `upstream_feedback` (e.g., `memory, upstream_feedback (external)`)
```
[the remark this answers is not in the document]
Comment (devseunggwan) [reply skills/retrospect/SKILL.md:967]: Declining this one — the leading space inside `` ` (external)` `` is intentional. The documented suffix is literally " (external)" with the leading space, as the adjacent example `memory, upstream_feedback (external)` shows; removing it would misdocument the suffix. This line is also pre-existing — this PR's hunk only changed the path on the line below — and the markdownlint CI check passes. Resolving as a false positive for this context.
